pkg/proxy/endpoints.go

```diff
@@ -407,20 +407,19 @@ func detectStaleConntrackEntries(oldEndpointsMap, newEndpointsMap EndpointsMap,
 		}
 
 		for _, ep := range epList {
-			// If the old endpoint wasn't Ready then there can't be stale
+			// If the old endpoint wasn't Serving then there can't be stale
 			// conntrack entries since there was no traffic sent to it.
-			if !ep.IsReady() {
+			if !ep.IsServing() {
 				continue
 			}
 
 			deleted := true
 			// Check if the endpoint has changed, including if it went from
-			// ready to not ready. If it did change stale entries for the old
+			// serving to not serving. If it did change stale entries for the old
 			// endpoint have to be cleared.
 			for i := range newEndpointsMap[svcPortName] {
 				if newEndpointsMap[svcPortName][i].String() == ep.String() &&
-					newEndpointsMap[svcPortName][i].IsReady() == ep.IsReady() &&
-					newEndpointsMap[svcPortName][i].GetIsLocal() == ep.GetIsLocal() {
+					newEndpointsMap[svcPortName][i].IsServing() == ep.IsServing() {
 					deleted = false
 					break
```

Comment:
You don't need to test Serving here; we already know that `ep` is Serving (line 412), and we already know that `newEndpointsMap[svcPortName][i]` is Serving (because it couldn't be in `newEndpointsMap` otherwise).